### PR TITLE
Merge common and specific code-templates

### PR DESCRIPTION
**Pull request by theory-in-progress**
### Description
Merge common and Specific code-templates

- Modified the code in fetchTemplates so that any live updates from vue in genCode is not overwritten
- Dynamically fetches the common code if the file exists, else proceeds with the code from specific file
- Created new function mergeCode to merge codes from specific and common	templates using ejs, the code tag being `#:::- replace_here :::#`
- The function takes in strings of specific file and common file and returns the rendered code using ejs
- The code tag is present in specific template which will be replaced with the code from common template

Feature Request

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [x] New feature
- [ ] Other

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -59,6 +59,14 @@ export function saveConfig(key, value) {
   }
 }
 
+// merges the code from the common and specific files using ejs
+function mergeCode(specificFileText, commonFileText) {
+  const replaced = ejs.render(specificFileText, {
+    replace_here: commonFileText
```
Comment (vfdev-5) [inline src/store.js:65]: Let's change the tag name to `from_template_common`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -109,4 +109,3 @@ jobs:
       - run: npm install -g pnpm
       - run: pnpm i --frozen-lockfile --color
       - run: pnpm lint
-      - run: python scripts/check_copies.py
```
Comment (vfdev-5) [inline .github/workflows/ci.yml:112]: please add steps to generate `./dist-tests` folder and to check if `./dist-tests` folder exists

File: src/templates/template-vision-segmentation/trainers.py [hunk 3]
```diff
@@ -3,7 +3,14 @@
...
 from data import prepare_image_mask
-from ignite.engine import DeterministicEngine, Engine, Events
+
+#::: if(it.deterministic) { :::#
+from ignite.engine import DeterministicEngine, Engine, Events  # usort: skip
+
+#::: } else { :::#
+from ignite.engine import Engine, Events
+
+#::: } :::#
```
Comment (vfdev-5) [inline src/templates/template-vision-segmentation/trainers.py:13]: @theory-in-progress please add return type hint to `setup_trainer` and `setup_evaluator`

Comment (theory-in-progress) [reply src/templates/template-vision-segmentation/trainers.py:13]: Should the changes to `setup_trainer` and `setup_evaluator` be in the same PR?

Comment (vfdev-5) [reply src/templates/template-vision-segmentation/trainers.py:13]: Just return type hints

**Pull request merged**